src/Maybe.js

```diff
@@ -29,11 +35,19 @@ Maybe.of = Maybe.Just;
 Maybe.prototype.of = Maybe.Just;
 
 Maybe.isJust = function(x) {
-  return x instanceof _Just;
+  if (typeof x.isJust !== 'boolean') {
```

Comment:
I think the `typeof` test is no longer necessary here (or in `isNothing`) and we can just return `x.isJust === true`;